### PR TITLE
podman events format json

### LINKED ISSUE #3159: podman events output jsonlines
/kind feature

**Description**

to make parsing `podman events` I need it to be in [jsonlines](http://jsonlines.org/) (aka ndjson)  

current output

```
$ podman events
2019-05-19 12:19:13.962020795 +0300 EEST container init 0c3b587013968ce5adc5f2dcbbcfd691c3f162bf43798f885d89769c68529d92 (image=docker.io/library/busybox:latest, name=my)
2019-05-19 12:19:14.086438455 +0300 EEST container start 0c3b587013968ce5adc5f2dcbbcfd691c3f162bf43798f885d89769c68529d92 (image=docker.io/library/busybox:latest, name=my)
2019-05-19 12:19:46.604928881 +0300 EEST container create 4871dacac25d09b95a00e3a72111c6813fa4e46941afdfe52b074c6e60f83072 (image=docker.io/library/busybox:latest, name=tender_mclaren)
2019-05-19 12:21:11.308193704 +0300 EEST container init 4871dacac25d09b95a00e3a72111c6813fa4e46941afdfe52b074c6e60f83072 (image=docker.io/library/busybox:latest, name=tender_mclaren)
2019-05-19 12:21:11.308565427 +0300 EEST container attach 4871dacac25d09b95a00e3a72111c6813fa4e46941afdfe52b074c6e60f83072 (image=docker.io/library/busybox:latest, name=tender_mclaren)
2019-05-19 12:21:11.357720034 +0300 EEST container start 4871dacac25d09b95a00e3a72111c6813fa4e46941afdfe52b074c6e60f83072 (image=docker.io/library/busybox:latest, name=tender_mclaren)
```

it seems there is no way to make it json

```
$ man podman events
...
--format
       Format the output using the given Go template.  An output value of json is not supported.
```

if you just dump a json per line with whatever internal format might work, but something like this would be better

```
{"time": 123, "kind": "container", "status":"start", "container_id":"123ee",  "from": "busybox", ...}
{"time": 123, "kind": "pod": "status":"start", "pod_id":"321aa", ...}
```

### PR DESCRIPTION
**Pull request by QiWang19**
Enable podman events to format the output as jsonlines
close #3159 

Signed-off-by: Qi Wang <qiwan@redhat.com>

### CODE REVIEW

**Comment by baude** [comment]
this function needs to be implemented for the local and remote client.  i believe you have only done the local client.

**Comment by rhatdan** [comment]
Podman events man page needs to be updated, as well as an example.

**Comment by QiWang19** [comment]
Implemented for the remote client and changes the man page. @baude @rhatdan  PTAL

**Comment by rhatdan** [comment]
LGTM
@baude @mheon @giuseppe @vrothberg @TomSweeneyRedHat PTAL

**Comment by baude** [comment]
@QiWang19 would it be possible to add a integration test to protect against regression?

File: test/e2e/events_test.go [hunk 1]
```diff
@@ -116,4 +116,14 @@ var _ = Describe("Podman events", func() {
 		Expect(result.ExitCode()).To(BeZero())
 	})
 
+	It("podman events format", func() {
+		Skip("need to verify images have correct packages for journald")
```
Comment (mheon) [inline test/e2e/events_test.go]: Unconditionally skipped?

Comment (baude) [reply test/e2e/events_test.go]: we should at least be able to do this if distro == fedora ... @QiWang19 there is a helper for this:
```
info := GetHostDistributionInfo()
info.Distribution
```

**Comment by QiWang19** [comment]
I saw this error `[+0658s] time="2019-06-05T17:17:58Z" level=error msg="unable to write pod event: "journal error: could not connect to journald socket""`  is it related with the skip? did I check the `info.Distribution != "fedora"` right?

**Comment by baude** [comment]
> Overall LGTM. It would be great to extend the tests to also check the content of the output. Some values might be different, but we could check for the keys.

I agree., this gives us a chance to test for specific events too, which is nice

**Pull request merged**